bootstrap.go

```diff
@@ -96,8 +96,26 @@ func startProfiler(options *options) error {
 }
 
 // stop with a graceful shutdown that includes flushing signals.
-func stop() error {
-	tracer.Stop()
-	profiler.Stop()
-	return nil
+func stop(options *options) error {
+	ctx := context.Background()
+
+	if options.stopTimeout != nil {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, *options.stopTimeout)
+		defer cancel()
+	}
+
+	errCh := make(chan error, 1)
```

Comment:
Since no error is returned by the go-routine you could simply defer the cancel in the go-routine and wait for the context to be done I think?

Reply:
Ah yes, I understand.